spec/string-class-spec.js

```diff
@@ -0,0 +1,152 @@
+
+    it('returns a string', () => {
+      expect(typeof string.toUpper()).toBe('string');
+    });
+
+    it('returns the string with all characters in upper case', () => {
+      expect(string.toUpper()).toEqual('NJERRYWERRY');
+      expect('NjerryWerry'.toUpper()).toEqual('NJERRYWERRY');
+      expect('Njerry Is Uppercase Now'.toUpper()).toEqual('NJERRY IS UPPERCASE NOW');
+      expect('njerry 123'.toUpper()).toEqual('NJERRY 123');
+    });
+  });
+
+  // toLower() tests
+  describe('toLower() Tests', () => {
+    const string = 'NJERRYWERRY';
+
+    it('returns a string', () => {
+      expect(typeof string.toLower()).toBe('string');
+    });
+
+    it('returns the string with all characters in lower case', () => {
+      expect(string.toLower()).toEqual('njerrywerry');
+      expect('NjerryWerry'.toLower()).toEqual('njerrywerry');
+      expect('Njerry Is LowerCase Now'.toLower()).toEqual('njerry is lowercase now');
+      expect('NJERRy 123'.toLower()).toEqual('njerry 123');
+    });
+  });
+
+  // ucFirst() tests
+  describe('ucFirst() Tests', () => {
+    const string = 'njerry';
+
+    it('returns a string', () => {
+      expect(typeof string.ucFirst()).toBe('string');
+    });
+
+    it('returns the string with the first character to upper case', () => {
+      expect(string.ucFirst()).toEqual('Njerry');
+      expect('njeRI'.ucFirst()).toEqual('NjeRI');
+      expect('njerry werry'.ucFirst()).toEqual('Njerry werry');
+    });
+  });
+
+  // isQuestion() tests
+  describe('isQuestion() Tests', () => {
+    it('returns a boolean value', () => {
+      expect(typeof ' '.isQuestion()).toBe('boolean');
+    });
+
+    it('returns true if string is a question', () => {
+      expect('Question?'.isQuestion()).toBe(true);
+    });
+
+    it('returns false if string is not a question', () => {
+      expect('Not a question'.isQuestion()).toBe(false);
+    });
+
+    it('returns false if question mark is not at the end of the string', () => {
+      expect('Invalid? Question'.isQuestion()).toBe(false);
+    });
+  });
+
+  // words() tests
+  describe('words() Tests', () => {
+    const string = 'array of words';
+
+    it('returns an array', () => {
+      expect(typeof string.words()).toBe('object');
+      expect(string.words() instanceof Array).toBe(true);
+      expect(Array.isArray(string.words())).toBe(true);
+    });
+
+    it('returns a list of words in the string as an Array', () => {
+      expect(string.words()).toEqual(['array', 'of', 'words']);
+      expect('array:of!words?is,here?'.words()).toEqual(['array', 'of', 'words', 'is', 'here']);
+      expect('?array::,*of!,.%^words'.words()).toEqual(['array', 'of', 'words']);
+      expect('don\'t go there!'.words()).toEqual(['don\'t', 'go', 'there']);
+    });
+  });
+
+  // wordCount() tests
+  describe('wordCount() Tests', () => {
+    const string = 'array of words';
+
+    it('returns a number', () => {
+      expect(typeof string.wordCount()).toBe('number');
```

Comment:
I do not think is a valid test.
Asserting that the word count returns a number does not give us any info that we need. It would essentially pass for an invalid number as well.